.gitignore

```diff
@@ -3,15 +3,17 @@
 /lib/
 /dist/
 /test-output/
-/temp*
 /.settings/
 /nbproject/private/
 /target/
 .classpath
 .project
 .checkstyle
-*.class
```

Comment:
I realize `.class` files are only built in the `target` directory, but we probably don't want to make these deletions anyways.  Changes to the `.gitignore` file should probably be part of a separate PR, as they are unrelated to these feature changes.